src/main/scala/mesosphere/marathon/core/task/update/impl/TaskStatusUpdateProcessorImpl.scala

```diff
@@ -72,18 +82,28 @@ class TaskStatusUpdateProcessorImpl @Inject() (
 object TaskStatusUpdateProcessorImpl {
   lazy val name = Names.named(getClass.getSimpleName)
 
+  /** Matches all states that are considered terminal for an unknown task */
+  private[impl] def terminalUnknown(status: MarathonTaskStatus): Boolean = status match {
+    case t: MarathonTaskStatus.Terminal => true
+    case MarathonTaskStatus.Unreachable => true
+    case _ => false
+  }
+
   private[this] val ignoreWhenUnknown = Set(
```

Comment:
I'd actually move this as `terminalStates` to `mesosphere.marathon.core.task.state.MarathonTaskStatus` since this will be a common thing and should be defined in one position.